// @arbitrum/orbit-sdk/dist/chains.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.nitroTestnodeL2 = exports.nitroTestnodeL1 = exports.arbitrumSepolia = exports.sepolia = exports.chains = void 0;
const viem_1 = require("viem");
const chains_1 = require("viem/chains");
Object.defineProperty(exports, "sepolia", { enumerable: true, get: function () { return chains_1.sepolia; } });
Object.defineProperty(exports, "arbitrumSepolia", { enumerable: true, get: function () { return chains_1.arbitrumSepolia; } });
const nitroTestnodeL1 = (0, viem_1.defineChain)({
    id: 1337,
    network: 'Nitro Testnode L1',
    name: 'nitro-testnode-l1',
    nativeCurrency: { name: 'Ether', symbol: 'ETH', decimals: 18 },
    rpcUrls: {
        default: {
            http: ['http://127.0.0.1:8545'],
        },
        public: {
            http: ['http://127.0.0.1:8545'],
        },
    },
    testnet: true,
});
exports.nitroTestnodeL1 = nitroTestnodeL1;
const nitroTestnodeL2 = (0, viem_1.defineChain)({
    id: 412346,
    network: 'Nitro Testnode L2',
    name: 'nitro-testnode-l2',
    nativeCurrency: { name: 'Ether', symbol: 'ETH', decimals: 18 },
    rpcUrls: {
        default: {
            http: ['http://127.0.0.1:8547'],
        },
        public: {
            http: ['http://127.0.0.1:8547'],
        },
    },
    testnet: true,
});
exports.nitroTestnodeL2 = nitroTestnodeL2;
exports.chains = [
    chains_1.sepolia,
    chains_1.arbitrumSepolia,
    nitroTestnodeL1,
    nitroTestnodeL2,
];
