// @arbitrum/orbit-sdk/dist/createRollupPrepareConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createRollupPrepareConfig = exports.defaults = void 0;
const viem_1 = require("viem");
const prepareChainConfig_1 = require("./prepareChainConfig");
const wasmModuleRoot = '0x0754e09320c381566cc0449904c377a52bd34a6b9404432e80afd573b67f7b17';
exports.defaults = {
    confirmPeriodBlocks: BigInt(150),
    extraChallengeTimeBlocks: BigInt(0),
    stakeToken: viem_1.zeroAddress,
    baseStake: (0, viem_1.parseEther)(String(0.1)),
    wasmModuleRoot,
    loserStakeEscrow: viem_1.zeroAddress,
    genesisBlockNum: BigInt(0),
    sequencerInboxMaxTimeVariation: {
        delayBlocks: BigInt(5760),
        futureBlocks: BigInt(48),
        delaySeconds: BigInt(86400),
        futureSeconds: BigInt(3600),
    },
};
function createRollupPrepareConfig({ chainConfig, ...params }) {
    return {
        ...exports.defaults,
        ...params,
        chainConfig: JSON.stringify(chainConfig ??
            (0, prepareChainConfig_1.prepareChainConfig)({
                chainId: Number(params.chainId),
                arbitrum: { InitialChainOwner: params.owner },
            })),
    };
}
exports.createRollupPrepareConfig = createRollupPrepareConfig;
