// @arbitrum/orbit-sdk/dist/createRollupPrepareTransaction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createRollupPrepareTransaction = void 0;
const viem_1 = require("viem");
const contracts_1 = require("./contracts");
function createRollupDecodeFunctionData(data) {
    return (0, viem_1.decodeFunctionData)({
        abi: contracts_1.rollupCreator.abi,
        data,
    });
}
function createRollupPrepareTransaction(tx) {
    return {
        ...tx,
        getInputs: function () {
            const { functionName, args } = createRollupDecodeFunctionData(tx.input);
            if (functionName !== 'createRollup') {
                throw new Error(`
          [createRollupPrepareTransaction] expected function name to be "createRollup" but got "${functionName}"
        `);
            }
            if (typeof args === 'undefined') {
                throw new Error(`[createRollupPrepareTransaction] failed to decode function data`);
            }
            return args;
        },
    };
}
exports.createRollupPrepareTransaction = createRollupPrepareTransaction;
