// @arbitrum/orbit-sdk/dist/prepareChainConfig.unit.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const vitest_1 = require("vitest");
const prepareChainConfig_1 = require("./prepareChainConfig");
const chainId = 69420;
const vitalik = '0xd8da6bf26964af9d7eed9e03e53415d37aa96045';
(0, vitest_1.it)('creates chain config with defaults', () => {
    const params = {
        chainId,
        arbitrum: {
            InitialChainOwner: vitalik,
        },
    };
    (0, vitest_1.expect)((0, prepareChainConfig_1.prepareChainConfig)(params)).toMatchSnapshot();
});
(0, vitest_1.it)('creates chain config with custom params', () => {
    const params = {
        chainId,
        homesteadBlock: 1,
        daoForkBlock: null,
        daoForkSupport: false,
        eip150Block: 1,
        eip150Hash: '0x1100000000000000000000000000000000000000000000000000000000000000',
        eip155Block: 1,
        eip158Block: 1,
        byzantiumBlock: 1,
        constantinopleBlock: 1,
        petersburgBlock: 1,
        istanbulBlock: 1,
        muirGlacierBlock: 1,
        berlinBlock: 1,
        londonBlock: 1,
        clique: {
            period: 1,
            epoch: 1,
        },
        arbitrum: {
            EnableArbOS: false,
            AllowDebugPrecompiles: true,
            DataAvailabilityCommittee: true,
            InitialArbOSVersion: 11,
            InitialChainOwner: vitalik,
            GenesisBlockNum: 1,
            MaxCodeSize: 40 * 1024,
            MaxInitCodeSize: 80 * 1024,
        },
    };
    (0, vitest_1.expect)((0, prepareChainConfig_1.prepareChainConfig)(params)).toMatchSnapshot();
});
