// @arbitrum/orbit-sdk/dist/prepareChainConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.prepareChainConfig = exports.defaults = void 0;
exports.defaults = {
    homesteadBlock: 0,
    daoForkBlock: null,
    daoForkSupport: true,
    eip150Block: 0,
    eip150Hash: '0x0000000000000000000000000000000000000000000000000000000000000000',
    eip155Block: 0,
    eip158Block: 0,
    byzantiumBlock: 0,
    constantinopleBlock: 0,
    petersburgBlock: 0,
    istanbulBlock: 0,
    muirGlacierBlock: 0,
    berlinBlock: 0,
    londonBlock: 0,
    clique: {
        period: 0,
        epoch: 0,
    },
    arbitrum: {
        EnableArbOS: true,
        AllowDebugPrecompiles: false,
        DataAvailabilityCommittee: false,
        InitialArbOSVersion: 10,
        GenesisBlockNum: 0,
        MaxCodeSize: 24576,
        MaxInitCodeSize: 49152,
    },
};
function prepareChainConfig(params) {
    return {
        ...exports.defaults,
        ...params,
        clique: { ...exports.defaults.clique, ...params.clique },
        arbitrum: { ...exports.defaults.arbitrum, ...params.arbitrum },
    };
}
exports.prepareChainConfig = prepareChainConfig;
