// @arbitrum/orbit-sdk/dist/setValidKeysetPrepareTransactionRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setValidKeysetPrepareTransactionRequest = void 0;
const ParentChain_1 = require("./types/ParentChain");
const setValidKeysetEncodeFunctionData_1 = require("./setValidKeysetEncodeFunctionData");
const upgradeExecutor_1 = require("./upgradeExecutor");
async function setValidKeysetPrepareTransactionRequest({ coreContracts, keyset, account, publicClient, }) {
    const chainId = publicClient.chain?.id;
    if (!(0, ParentChain_1.validParentChainId)(chainId)) {
        throw new Error('chainId is undefined');
    }
    const request = await publicClient.prepareTransactionRequest({
        chain: publicClient.chain,
        to: coreContracts.upgradeExecutor,
        data: (0, upgradeExecutor_1.upgradeExecutorEncodeFunctionData)({
            functionName: 'executeCall',
            args: [
                coreContracts.sequencerInbox,
                (0, setValidKeysetEncodeFunctionData_1.setValidKeysetEncodeFunctionData)(keyset), // targetCallData
            ],
        }),
        account,
    });
    return { ...request, chainId };
}
exports.setValidKeysetPrepareTransactionRequest = setValidKeysetPrepareTransactionRequest;
